packages/components/ovh-shell/src/shell/index.ts

```diff
@@ -1,22 +1,39 @@
+import { Environment, fetchConfiguration } from '@ovh-ux/manager-config';
+
 import Shell from './shell';
 import IFrameMessageBus from '../message-bus/iframe';
 import DirectClientMessageBus from '../message-bus/direct-client';
 
-export function initShell() {
-  const shell = new Shell(new DirectClientMessageBus());
-  return {
-    emitEvent: (eventId: string, data: unknown): void =>
-      shell.emitEvent(eventId, data),
-    registerPlugin: (
-      pluginId: string,
-      pluginApi: Record<string, CallableFunction>,
-    ) => shell.getPluginManager().registerPlugin(pluginId, pluginApi),
-    setPluginAvailability: (pluginId: string, availability: boolean) =>
-      shell.getPluginManager().setPluginAvailability(pluginId, availability),
-    setIframeMessageBus: (iframe: HTMLIFrameElement) =>
-      shell.setMessageBus(new IFrameMessageBus(iframe)),
-    i18n: () => shell.getPluginManager().plugins.i18n.instance,
-  };
+interface IShell {
+  getEnvironment: () => Environment;
+  emitEvent: (eventId: string, data: unknown) => void;
+  registerPlugin: (
+    pluginId: string,
+    pluginApi: Record<string, CallableFunction>,
+  ) => void;
+  setPluginAvailability: (pluginId: string, availability: boolean) => void;
+  setIframeMessageBus: (iframe: HTMLIFrameElement) => void;
+  i18n: () => unknown;
```

Comment:
I think this is of type `Record<string, CallableFunction>`. 

It comes from `shell.getPluginManager().plugins.i18n.instance` , `i18n` being of type `PluginEntry` which has the `instance` object in it.